### DISCUSSION TITLE
Configuring WiFi with a combined 2.4/5G AP Point

### DISCUSSION THREAD

**Discussion by pjschaffer** [Resinchem/ESP-Parking-Assistant#30] [Troubleshooting and Assistance]
Just wanted to share an issue I ran across and how I resolved it.

Issue: Can't configure WiFi with a combined 2.4/5G AP Point (ASUS Router)

Background: My normal configuration for my WiFi AP is a combined 2.4/5G SSID where both networks share the same SSID and login credentials.  I recently updated my two Parking Assistants from v0.44 to v0.46.  One of the two updated just fine but the second one had some fatal failure and could not be reflashed even with a cable.  That said, I built another Parking Assistant controller and proceeded to try to add it to my network.  I attempted with several different devices, a phone, tablet and finally a laptop with no luck.  I could add my credentials but the device would never connect to the AP.

Solution (at least the one I found): I disabled the combined AP on my router and separated the 2.4G band on it's own SSID (the same as my normal SSID name).  Once I did this, the controller connected to my AP on the first try and I was then able to configure it as before.  I then recombined the 2.4 and 5G bands on the router and the controller remained connected.  I'm not sure id the issue was my network configuration or had something to do with the ESP8266 wifi chip, but either way, it took longer to troubleshoot this issue then it did to build a new controller.  Hopefully this can save somebody some time in the future.

**Comment by Resinchem** [comment]
Thanks for the info... and the workaround that you found.  While it could be related to your network or issues with the ESP8266, the wifi library I'm using has its own issues.... known problems include handling some special characters in the SSID or password, truncating very long SSIDs, etc.  I selected it because of the availability of the onboarding process.  There have been some additional releases of this library, but I'm currently stuck on an older version because later versions require using LittleFS and I'm using SPIFFS.  Migrating that is something that I need to work on and fully test so that settings are not lost on future firmware updates.

I've also developed my own version of the wifi library with onboarding, and while it works for my particular network/SSID, it needs substantially more testing on other networks before it would be ready to release.  I'd also like to update a number of other libraries and move the development environment to IDE 2.x (instead of the original IDE 1.8.x).  So... yes, a lot of updates on the board that I'd like to roll into a future firmware update.  But replacing the existing wifi library is near the top.  But doing that requires me to make other upgrades as well.

Regardless, thanks for sharing your issue.. and how you worked around it.  Hopefully it can help out someone else... until I get get some of these other updates done.
